src/GitReleaseManager.Core/Provider/GitHubProvider.cs

```diff
@@ -25,13 +30,66 @@ public class GitHubProvider : IVcsProvider
         private const int PAGE_SIZE = 100;
         private const string NOT_FOUND_MESSGAE = "NotFound";
 
+        // This query fragment will be executed for issues and pull requests
+        // because we don't know whether issueNumber refers to an issue or a PR
+        private const string CONNECT_AND_DISCONNECT_EVENTS_GRAPHQL_QUERY_FRAGMENT = @"
+		{0}(number: $issueNumber) {{
+			timelineItems(first: $pageSize, itemTypes: [CONNECTED_EVENT, DISCONNECTED_EVENT]) {{
+				nodes {{
+					__typename,
+					...on ConnectedEvent {{
+						createdAt,
+						id,
+						source {{
+							__typename,
+							... on Issue {{
+  								number
+							}}
+							... on PullRequest {{
+  								number
+							}}
+						}},
+						subject {{
+							__typename 
+							... on Issue {{
+  								number
+							}}
+							... on PullRequest {{
+  								number
+							}}
+						}}
+					}}
+					...on DisconnectedEvent {{
+						createdAt,
+					}}
+				}}
+			}}
+		}}";
+
+        private const string CONNECT_AND_DISCONNECT_EVENTS_GRAPHQL_QUERY = @"
+query ConnectAndDisconnectEvents($repoName: String!, $repoOwner: String!, $issueNumber: Int!, $pageSize: Int!) {{
+	repository(name: $repoName, owner: $repoOwner) {{
+		{0},
+		{1}
+	}}
+}}";
+
         private readonly IGitHubClient _gitHubClient;
         private readonly IMapper _mapper;
+        private readonly IGraphQLClient _graphQLClient;
 
         public GitHubProvider(IGitHubClient gitHubClient, IMapper mapper)
         {
             _gitHubClient = gitHubClient;
             _mapper = mapper;
+
+            var graphQLClient = new GraphQLHttpClient(new GraphQLHttpClientOptions { EndPoint = new Uri("https://api.github.com/graphql") }, new SystemTextJsonSerializer());
```

Comment:
Instantiating the GraphQLHttpClient here doesn't seem right. This client should probably be injected in the constructor or maybe, rather than mixing the REST GitHub client with the graph QL client in the same provider, there should be a separate provider specifically for graphQL operations.

@gep13 what do you think?